platform-sdk/docs/proposals/state-lifecycle/proposal.md

```diff
@@ -33,66 +33,76 @@ These methods correspond to the state lifecycle as follows:
-    - Sealing consensus rounds.
-    - Keeping the `freeze time` parameter up-to-date.
-    - Providing state for signing.
-    - Initializing the state by extracting it from the signed state.
+- Keeping track of the references to the latest immutable state and the latest mutable state. This includes creating a copy of the state and updating the references.
+This functionality belongs to `swirlds-state-api`.
+
+- Handling Platform-related events (belongs to `swirlds-platform-core`):
+  - Handling consensus rounds.
+  - Sealing consensus rounds.
+  - Keeping the `freeze time` parameter up-to-date.
+  - Providing state for signing.
+  - Initializing the state by extracting it from the signed state.
 
 These two sets of responsibilities should be separated. The `swirlds-state-impl` should not have any details related to the platform. However, it should provide all necessary levers for the platform to interact with the state and its lifecycle.
 
-Therefore, `swirlds-state-api` needs a `StateManager` interface with an implementation `StateManagerImpl` in `swirlds-state-impl`  that will have the following responsibilities:
-- Create an empty state
-- Keeping track of the references to the latest immutable state and the latest mutable state.
-- Tracking states that existed before the latest immutable state. These should be available for lookup by their round number.
-- Restricts mutability to a single state. That is, only one state object should be mutable at any given time.
-- Evicting old states from memory that are no longer needed. `evictState` method *must* fail if the state is still mutable.
-- Loading snapshots from the disk.
+Therefore, `swirlds-state-api` needs a `StateLifecycleManager` interface with an implementation `StateLifecycleManagerImpl` in `swirlds-state-impl`  that will have the following responsibilities:
+- Take initial state as a starting point. It can be a genesis state or a state loaded from a snapshot.
+- Keep track of the references to the latest immutable state and the latest mutable state.
+- Restrict mutability to a single state. That is, only one state object should be mutable at any given time.
+
+The interface will look something like this:
+
+```java
+public interface StateLifecycleManager {
+    void setInitialState(State initialState);
+
+    State getLatestImmutableState();
 
-Additionally, there should be a configuration parameter that limits the number of states kept in memory. When the limit is reached, the oldest state should be evicted.
-Currently, we rely on the implicit garbage collection. We have a reference count mechanism to achieve this.
-We want to get rid of this mechanism and rely on the explicit eviction mechanism.
+    State getMutableState();
+}
+```
+
+**Imporatant**: It's up to the client code to make sure that the state is released once it's no longer needed.
+Without this, the state will not be evicted from memory and eventually the application will run out of memory.
```

Comment:
Two comments:
1. Typo in "Important"
2. Nothing about reservations has been explained in this document so far so this note is not meaningful. You don't need to go into the details of how it works, but a short paragraph explaining that reference counts are used to determine when a state is eligible for GC would be helpful. Then you could also state that the state module will ensure the mutable and latest immutable states are held onto, but if the user of the library needs to keep a state around longer than that, it is their responsibility to ensure proper use of additional reservations. The CN will do with with the existing class structures of SignedState and ReservableSignedState that sit on top of these state reservations.

Reply:
Sure, I will add information about reservations.